test/attributes.js

```diff
@@ -41,3 +41,40 @@ QUnit.test( ".removeAttr( boolean attribute )", function( assert ) {
 	} );
 
 } );
+
+QUnit.test( "", function( assert ) {
+	assert.expect( 13 );
+
+	var e = jQuery( "<div />" ).appendTo( "#qunit-fixture" );
+
+	expectWarning( "toggling initially empty class", function() {
+		e.toggleClass( true );
+		assert.equal( e[ 0 ].className, "", "Assert class is empty (data was empty)" );
+	} );
+
+	expectNoWarning( "toggleClass not full className", function() {
+		e.attr( "class", "" );
+		e.toggleClass( "classy" );
+		assert.equal( e.attr( "class" ), "classy", "class was toggle-set" );
+		e.toggleClass( "classy", false );
+		assert.equal( e.attr( "class" ), "", "class was toggle-removed" );
+	} );
+
+	expectWarning( "toggleClass full-className manipulations", function() {
+		e.addClass( "testD testE" );
+		assert.ok( e.is( ".testD.testE" ), "Assert class present" );
+		e.toggleClass();
+		assert.ok( !e.is( ".testD.testE" ), "Assert class not present" );
```

Comment:
These tests were copied from core pretty much as-is so I hesitate to add more assertions. Whatever it was doing before was lightly documented and we're deprecating it now.